api/base/pagination.py

```diff
@@ -89,6 +93,21 @@ def get_paginated_response(self, data):
 
         response_dict = self.get_response_dict(data, reversed_url)
 
+        if view_name == 'nodes:node-comments' and self.request.query_params.get('related_counts', False):
```

Comment:
This special-casing of one view looks suspicious. Could you use a subclass instead?

```python
class CommentPagination(JSONAPIPagination):
    def get_paginated_response(...):
        #...

class NodeCommentsList(...):
    pagination_class = CommentPagination
   #...
```